controllers/config/manifest.go

```diff
@@ -64,9 +64,12 @@ func Delete(cl client.Client, owner metav1.Object, templatePath string, context
 	if owner != nil {
 		asMfOwner := owner.(mf.Owner)
 		fns = append(fns, mf.InjectOwner(asMfOwner))
-		fns = append(fns, mf.InjectNamespace(asMfOwner.GetNamespace()))
+		if asMfOwner.GetNamespace() != "" {
+			fns = append(fns, mf.InjectNamespace(asMfOwner.GetNamespace()))
+		} else {
+			fns = append(fns, mf.InjectNamespace(namespace))
+		}
```

Comment:
```suggestion
		fns = append(fns, mf.InjectNamespace(namespace))
```
@chinhuang007 no need for the if/else here